### ISSUE TITLE
[Chatllama] looks like REWARD MODEL donesn't support OPT

### ISSUE THREAD

**Issue by balcklive** [nebuly-ai/nebullvm#281]
here's the piece of code of reward.py:
`class RewardModel(torch.nn.Module):
    """Model to be trained to predict the reward for RL.
    or to be used as Critic in RL.

    Attributes:
        model (torch.nn.Module): Model to be used for the reward model
        tokenizer (torch.nn.Module): Tokenizer to be used for the reward model
        head (torch.nn.Module): Head to be used for the reward model
        config (ConfigReward): Config parameters for the reward model
        max_model_tokens (int): Maximum sequence length for the reward model

    Methods:
        forward: Forward pass of the model (used by the critic)
        save: Save the model
        load: Load the model
        get_reward: Get the reward for a given input (used by the reward model)
    """

    def __init__(self, config: ConfigReward) -> None:
        super().__init__()
        # load the model -- add here other models
        head_hidden_size = config.model_head_hidden_size
        if config.model in hf_models:
            self.tokenizer = AutoTokenizer.from_pretrained(
                config.model,
                padding_side="left",
                truncation_side="left",
            )
            class RewardModel(torch.nn.Module):
    """Model to be trained to predict the reward for RL.
    or to be used as Critic in RL.

    Attributes:
        model (torch.nn.Module): Model to be used for the reward model
        tokenizer (torch.nn.Module): Tokenizer to be used for the reward model
        head (torch.nn.Module): Head to be used for the reward model
        config (ConfigReward): Config parameters for the reward model
        max_model_tokens (int): Maximum sequence length for the reward model

    Methods:
        forward: Forward pass of the model (used by the critic)
        save: Save the model
        load: Load the model
        get_reward: Get the reward for a given input (used by the reward model)
    """

    def __init__(self, config: ConfigReward) -> None:
        super().__init__()
        # load the model -- add here other models
        head_hidden_size = config.model_head_hidden_size
        if config.model in hf_models:
            self.tokenizer = AutoTokenizer.from_pretrained(
                config.model,
                padding_side="left",
                truncation_side="left",
            )
            self.model = AutoModel.from_pretrained("gpt2-large") ###<------watch this line!!!!!!!!!!!!!!!!!!!!!!
            self.tokenizer.pad_token = self.tokenizer.eos_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(self.model.config.n_embd, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        elif config.model == "bart-base":
            bart_config = BartConfig.from_pretrained("facebook/bart-base")
            bart_config.max_position_embeddings = 2048 + 1024
            self.model = BartModel(bart_config)
            self.tokenizer = BartTokenizer.from_pretrained(
                "facebook/bart-large",
                padding_side="left",
                truncation_side="left",
            )
            self.tokenizer.pad_token = self.tokenizer.eos_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(768, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        elif config.model == "longformer-base-4096":
            self.model = AutoModel.from_pretrained(
                "allenai/longformer-base-4096"
            )
            self.tokenizer = AutoTokenizer.from_pretrained(
                "allenai/longformer-base-4096",
                padding_side="left",
                truncation_side="left",
            )
            self.tokenizer.eos_token = self.tokenizer.pad_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(768, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        else:
            raise ValueError(f"model {config.model} not supported")
        # store config
        self.config = config
        if os.path.exists(config.model_folder) is False:
            os.mkdir(config.model_folder)
        else:
            self.load()
        # freeze model parameters (only train the head)
        for param in self.model.parameters():
            param.requires_grad = False
        # move model to device
        # self.model.to(config.device)
        # self.head.to(config.device)
            self.tokenizer.pad_token = self.tokenizer.eos_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(self.model.config.n_embd, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        elif config.model == "bart-base":
            bart_config = BartConfig.from_pretrained("facebook/bart-base")
            bart_config.max_position_embeddings = 2048 + 1024
            self.model = BartModel(bart_config)
            self.tokenizer = BartTokenizer.from_pretrained(
                "facebook/bart-large",
                padding_side="left",
                truncation_side="left",
            )
            self.tokenizer.pad_token = self.tokenizer.eos_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(768, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        elif config.model == "longformer-base-4096":
            self.model = AutoModel.from_pretrained(
                "allenai/longformer-base-4096"
            )
            self.tokenizer = AutoTokenizer.from_pretrained(
                "allenai/longformer-base-4096",
                padding_side="left",
                truncation_side="left",
            )
            self.tokenizer.eos_token = self.tokenizer.pad_token
            self.head = torch.nn.Sequential(
                torch.nn.Linear(768, head_hidden_size),
                torch.nn.ReLU(),
                torch.nn.Linear(head_hidden_size, 1),
                Rearrange("... 1 -> ..."),
            )
        else:
            raise ValueError(f"model {config.model} not supported")
        # store config
        self.config = config
        if os.path.exists(config.model_folder) is False:
            os.mkdir(config.model_folder)
        else:
            self.load()
        # freeze model parameters (only train the head)
        for param in self.model.parameters():
            param.requires_grad = False
        # move model to device
        self.model.to(config.device)
        self.head.to(config.device)`

the marked line says, even if you specify a reward model type like opt-1.3b, the code still gives you a gpt2 model. Is this a bug or on purpose?

**Comment by PierpaoloSorbellini** [comment]
Hi @balcklive. Thanks for opening this issue! It is a typo that has already been fixed in the PR #233. You should checkout to that branch for now. We will merge into the main branch at the starting of next week or earlier. 
Have nice day :)

**Comment by balcklive** [comment]
OK， I see it, this issue is closed
